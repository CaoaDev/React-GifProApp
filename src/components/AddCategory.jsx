import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setAnime } ) => {
    const [inputValue, setImputValue] = useState( '' )
    const handleInputChange = ( e ) => {
        setImputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        console.log( 'handleSubmit', inputValue );
        if ( inputValue.trim().length >= 1 ) {
            setAnime ( anime => [ inputValue, ...anime ] );
            setImputValue ('');
        }
    }
    // evento para limpiar imput por onclick
    // const clean = (e) => {
    //     e.preventDefault();
    //     setImputValue('')
    // }
    // funcion para agregar una lista en el setAnimes
    // const handleAdd = () => {
    //     setAnimes( animes => [...animes, 'HunterxHunter', 'Madoka Magica', 'Kimetsu'] );
    // }

    return (
        <form onSubmit={ handleSubmit }>
            <p> { inputValue } </p>    
            <h1>Agregando: { inputValue }</h1>
             < input
             type= 'text'
             value={ inputValue }
             onChange={ handleInputChange }
            //  onClick={ clean }
             placeholder='Nuevo Anime... '
             />
             </form>
             )
            }
            
            AddCategory.propTypes = {
                setAnime: PropTypes.func.isRequired
            }