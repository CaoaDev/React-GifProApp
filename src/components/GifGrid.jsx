import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
// import { getgif } from '../helpers/getGifs';

export const GifGrid = ( { anime } ) => {
  const { images, loading } = useFetchGifs( anime );

  // const [images, setImages] = useState([]);
  // useEffect( () => {
  //   getgif( anime )
  //   .then( setImages );
  // }, [ anime ]);

  return (
    <div>
      <h3 className= 'animate__bounceOutDown'>{ anime } </h3> 
      { loading && <p className= 'animate__fadeIn'>Loading</p>}
      <div className='card-grid'>
        {
          images.length > 0
            ?
              images.map ( ( img )=>(
                <GifGridItem 
                  key = { img.id }
                  { ...img }
                />
              ))
            :
              <p>no se encontro informacion de { anime }</p>
        }
      </div>
    </div>
  )
};

GifGrid.prototype = {
  category: PropTypes.string.isRequired
}