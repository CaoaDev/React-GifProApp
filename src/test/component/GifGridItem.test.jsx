import react  from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

  const title = 'Un titulo';
  const url = 'https://Una_Url';
  const wrapper = shallow ( <GifGridItem title= { title } url= { url } /> )

  test('Debe mostrar el componenete correctamente', () => {
      expect ( wrapper ).toMatchSnapshot();
  });

  test('Debe tener un parrafo en el title', () => {
    const p = wrapper.find ( 'p' );
    expect ( p.text().trim() ).toBe ( title ); 
  });

  test('La imagen debe tener de props la url y alt', () => {
     const img = wrapper.find ( 'img' );
     console.log( img.html() );
     console.log( img.props() );
     console.log( img.prop('alt') );
     console.log( img.prop('src') );
     console.log( img.props().alt );
     console.log( img.props().src );
     // diferentes formas de ver las props, si tienes enzyne puedes usar props, si no html
     expect( img.prop( 'src' ) ).toBe ( url );
     expect( img.prop( 'alt' ) ).toBe ( title );
     console.log(img.prop('src'),' es igual a: ',url);
     console.log(img.prop('alt'),' es igual a: ',title);
     
  });

  test('Debe de tener animate__bounceOut', () => {
    const div = wrapper.find('div');
    console.log( div.prop('className'));
    // con el console log podemos ver las props de la const
    const className = div.prop('className');
    expect ( className.includes( 'animate__bounceOut' ) ).toBe(true);
    expect ( className.includes( 'animate__bounceOut' ) ).not.toBe(false);

  });
  
})