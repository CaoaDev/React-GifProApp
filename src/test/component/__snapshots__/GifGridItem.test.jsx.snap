// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componenete correctamente 1`] = `
<div
  className="card animate__bounceOut"
>
  <img
    alt="Un titulo"
    src="https://Una_Url"
  />
  <p>
    Un titulo
  </p>
</div>
`;
